Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -459,22 +459,36 @@ def __lt__(self, other):
 		if type(self) != type(other):
 			return NotImplemented
 
-		# implemented so that list.sort() sorts according to the spec.
-		selfTuple = (
-			getattr(self, "platformID", None),
-			getattr(self, "platEncID", None),
-			getattr(self, "langID", None),
-			getattr(self, "nameID", None),
-			getattr(self, "string", None),
-		)
-		otherTuple = (
-			getattr(other, "platformID", None),
-			getattr(other, "platEncID", None),
-			getattr(other, "langID", None),
-			getattr(other, "nameID", None),
-			getattr(other, "string", None),
-		)
-		return selfTuple < otherTuple
+		try:
+			# implemented so that list.sort() sorts according to the spec.
+			selfTuple = (
+				getattr(self, "platformID", None),
+				getattr(self, "platEncID", None),
+				getattr(self, "langID", None),
+				getattr(self, "nameID", None),
+				self.toBytes(),
+			)
+			otherTuple = (
+				getattr(other, "platformID", None),
+				getattr(other, "platEncID", None),
+				getattr(other, "langID", None),
+				getattr(other, "nameID", None),
+				other.toBytes(),
+			)
+			return selfTuple < otherTuple
+		except UnicodeEncodeError:
```

Comment:
Maybe also catch `AttributeError`, since `toBytes` unconditionally gets the `string` attribute, whereas the others are (maybe too cautiously) using `gettattr(self, ..., None)`
actually, let's turn all these `getattr` to unconditional self.platformID, self.platEncID, etc. and catch AttributeError and return NotImplemented if any of these attributes is missing. This can only happen for 1) an object that is not a NameRecord, or an unlikely incomplete NameRecord object which has not been fully populated. In both cases the best is to return NotImplemented.
I also think the warning is too detailed and verbose for an occurrance which is not very likely to ever occur in real world. It will raise an UnicodeEncodeError _anyway_ as soon as you attempt to compile that thing, and it's clear where the error is coming from. Just remove the logging.warning, thanks